pgml-extension/pgml_rust/src/orm/estimator.rs

```diff
@@ -431,3 +432,82 @@ impl Estimator for BoosterBox {
         self.contents.predict(&features).unwrap()[0]
     }
 }
+
+/// A wrapper around a Scikit estimator.
+/// The estimator is a Python object and can only be used
+/// inside Python::with_gil.
+pub struct SklearnBox {
+    pub contents: Box<Py<PyAny>>,
+}
+
+impl Drop for SklearnBox {
+    fn drop(&mut self) {
+        // I don't think this works because drop for self must
+        // be executed before the drop of fields?
+        Python::with_gil(|_py| {});
+    }
+}
+
+impl SklearnBox {
+    pub fn new(contents: Py<PyAny>) -> SklearnBox {
+        SklearnBox {
+            contents: Box::new(contents),
+        }
+    }
+}
+
+impl std::ops::Deref for SklearnBox {
+    type Target = Py<PyAny>;
+
+    fn deref(&self) -> &Self::Target {
+        self.contents.as_ref()
+    }
+}
+
+impl std::ops::DerefMut for SklearnBox {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        self.contents.as_mut()
+    }
+}
+
+unsafe impl Send for SklearnBox {}
+unsafe impl Sync for SklearnBox {}
+
+impl std::fmt::Debug for SklearnBox {
+    fn fmt(
+        &self,
+        formatter: &mut std::fmt::Formatter<'_>,
+    ) -> std::result::Result<(), std::fmt::Error> {
+        formatter.debug_struct("SklearnBox").finish()
+    }
+}
+
+impl serde::Serialize for SklearnBox {
+    fn serialize<S>(&self, _serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        unreachable!()
```

Comment:
should this be `todo!()` or is it really unreachable!()? Maybe we should modify the upstream interface, I mean trait definition...

Reply:
Definitely maybe a todo